### ISSUE TITLE
Would you leverage the native support of upsert in PostgreSQL 9.5?

### ISSUE THREAD

**Issue by coin8086** [seamusabshere/upsert#78]
Now PostgreSQL 9.5 natively supports upsert:

```
The syntax is INSERT ... ON CONFLICT DO NOTHING/UPDATE. This is the Postgres implementation of the popular UPSERT command.
```

Here's the release notes:

https://www.postgresql.org/docs/9.5/static/release-9-5.html

**Comment by pnomolos** [comment]
I just began working on this.  The major problem (from the perspective of this project) is that the PostgreSQL implementation requires there to be a unique constraint on the columns being used in the `ON CONFLICT` clause.

"upsert" shouldn't be creating these constraints.  The best case scenario is probably the following:

- Is the version >= 9.5 __and__ does a constraint across all the selector fields?  Use `INSERT ... ON CONFLICT`.
- Else use existing function.

**Comment by pnomolos** [comment]
@coin8086 See https://github.com/seamusabshere/upsert/pull/79 - hopefully it can be added before too much longer :)

**Comment by pnomolos** [comment]
Closed via #79.  Will be used if it's available (dependent on PG version as well as existence of a unique index on the column(s) being used for selectors).

**Comment by coin8086** [comment]
Thanks!
